cli/flags.rs

```diff
@@ -197,197 +132,205 @@ fn create_cli_app<'a, 'b>() -> App<'a, 'b> {
-      let files: Vec<String> = fmt_match
-        .values_of("files")
-        .unwrap()
-        .map(String::from)
-        .collect();
-      rest_argv.extend(files);
-    }
-    (script, Some(script_match)) => {
-      rest_argv.extend(vec![script.to_string()]);
-      // check if there are any extra arguments that should
-      // be passed to script
-      if script_match.is_present("") {
-        let script_args: Vec<String> = script_match
-          .values_of("")
-          .unwrap()
-          .map(String::from)
-          .collect();
-        rest_argv.extend(script_args);
-      }
-    }
-    _ => {}
+  if matches.is_present("log-debug") {
+    flags.log_debug = true;
+  }
+  if matches.is_present("version") {
+    flags.version = true;
+  }
+  if matches.is_present("reload") {
+    flags.reload = true;
+  }
+  if matches.is_present("allow-read") {
+    flags.allow_read = true;
+  }
+  if matches.is_present("allow-write") {
+    flags.allow_write = true;
+  }
+  if matches.is_present("allow-net") {
+    flags.allow_net = true;
+  }
+  if matches.is_present("allow-env") {
+    flags.allow_env = true;
+  }
+  if matches.is_present("allow-run") {
+    flags.allow_run = true;
+  }
+  if matches.is_present("allow-high-precision") {
+    flags.allow_high_precision = true;
+  }
+  if matches.is_present("allow-all") {
+    flags.allow_read = true;
+    flags.allow_env = true;
+    flags.allow_net = true;
+    flags.allow_run = true;
+    flags.allow_read = true;
+    flags.allow_write = true;
+    flags.allow_high_precision = true;
+  }
+  if matches.is_present("no-prompt") {
+    flags.no_prompts = true;
   }
-
   if matches.is_present("v8-options") {
-    // display v8 help and exit
-    // TODO(bartlomieju): this relies on `v8_set_flags` to swap `--v8-options` to help
-    v8_set_flags(vec!["deno".to_string(), "--v8-options".to_string()]);
+    flags.v8_help = true;
   }
-
   if matches.is_present("v8-flags") {
-    let mut v8_flags: Vec<String> = matches
+    let v8_flags: Vec<String> = matches
       .values_of("v8-flags")
       .unwrap()
       .map(String::from)
       .collect();
 
-    v8_flags.insert(1, "deno".to_string());
-    v8_set_flags(v8_flags);
+    flags.v8_flags = Some(v8_flags);
   }
 
-  let flags = DenoFlags::from(matches);
-  Ok((flags, rest_argv))
+  flags
 }
 
-#[test]
-fn test_set_flags_1() {
-  let (flags, rest) = set_flags(svec!["deno", "--version"]).unwrap();
-  assert_eq!(rest, svec!["deno"]);
-  assert_eq!(
-    flags,
-    DenoFlags {
-      version: true,
-      ..DenoFlags::default()
-    }
-  );
-}
+#[cfg(test)]
+mod tests {
+  use super::*;
 
-#[test]
-fn test_set_flags_2() {
-  let (flags, rest) =
-    set_flags(svec!["deno", "-r", "-D", "script.ts"]).unwrap();
-  assert_eq!(rest, svec!["deno", "script.ts"]);
-  assert_eq!(
-    flags,
-    DenoFlags {
-      log_debug: true,
-      reload: true,
-      ..DenoFlags::default()
-    }
-  );
-}
+  fn flags_from_vec(args: Vec<String>) -> DenoFlags {
```

Comment:
I feel like this function should be the high level public “parse_flags” while the other function which is currently called “parse_flags” should be either combined into this or called something else (and be internal to this module)

Reply:
Yeah, I had similar feeling about it, but that means we no longer have `matches` var available in `main()`, that gives nice switch on `matches.subcommand()`.

Reply:
Alternatively `parse_flags` may return `DenoFlags, ArgMatches` to handle both cases